core/src/main/java/io/grpc/util/AdvancedTlsX509TrustManager.java

```diff
@@ -49,23 +50,22 @@
   private static final Logger log = Logger.getLogger(AdvancedTlsX509TrustManager.class.getName());
 
   private final Verification verification;
-  private final PeerVerifier peerVerifier;
   private final SslSocketAndEnginePeerVerifier socketAndEnginePeerVerifier;
 
   // The delegated trust manager used to perform traditional certificate verification.
   private volatile X509ExtendedTrustManager delegateManager = null;
 
-  private AdvancedTlsX509TrustManager(Verification verification,  PeerVerifier peerVerifier,
+  private AdvancedTlsX509TrustManager(Verification verification,
       SslSocketAndEnginePeerVerifier socketAndEnginePeerVerifier) throws CertificateException {
     this.verification = verification;
-    this.peerVerifier = peerVerifier;
     this.socketAndEnginePeerVerifier = socketAndEnginePeerVerifier;
   }
 
   @Override
   public void checkClientTrusted(X509Certificate[] chain, String authType)
       throws CertificateException {
-    checkTrusted(chain, authType, null, null, false);
+    throw new CertificateException(
+        "Either SSLEngine or Socket should be available. Consider upgrading your java version");
```

Comment:
Upgrading Java is unlikely to help; it'd have to be some library or application calling the old methods. Let's say, "Not enough information to validate peer. SSLEngine or Socket required."